### DISCUSSION TITLE
call getInitialProps in loadMore of infinite scroll

### DISCUSSION THREAD

**Discussion by farouk46015** [vercel/next.js#12478] [Help]
In my application i am trying to get consecutive  parts of data by calling pagination in backend 
i am using  infinite scroll to call 
 
fetch data of component in this way 
`  static async getInitialProps({store , isServer , pathname , query }){
    await modules.get('pages',{page: '1' ,size: '7' })
    const { data ,  pages , current , ...rest } = store.getState().main
    return { data , pages , current }
  }`

first can i call getInitialProps on function loadMore of infinite scroll 

then i need to return data from getInitialProps every time

**Comment by rafaelalmeidatk** [comment]
`getInitialProps` is only for the initial data. After the page is rendered, you need to do the subsequent fetch calls in the component, then updating some state to render more results

**Reply by exbotanical** [reply]
I'm about to implement this as well. @rafaelalmeidatk So what you are suggesting is we fetch the first 'page'/chunk of data in `getInitialProps` and proceed to make `fetch` calls in `loadMore` as needed? (As opposed to fetching _all_ of the data in getInitialProps, then mounting only _n_ items per page/load)

**Reply by rafaelalmeidatk** [reply]
Yeah, if you have a lot of data and fetch it all in `getInitialProps`, you will deliver way more data than the necessary to initially render the page

**Comment by Ehab97** [comment]
did find solution or idea ?
